cmd/event-received/main.go

```diff
@@ -77,6 +78,28 @@ type uidEvent struct {
 	UID string `json:"uid"`
 }
 
+type feeApprovedEvent struct {
+	UID          string      `json:"uid"`
+	ApprovedType pay.FeeType `json:"approvedType"`
+}
+
+func (e *feeApprovedEvent) UnmarshalJSON(data []byte) error {
+	var v struct{ UID, ApprovedType string }
```

Comment:
My bad, I'd assumed as we were getting the string rather than the int we'd need the manual approach. Is this Go being helpful or is the enum package that's doing the magic?

Reply:
`enumerator` generates an `UnmarshalText` method